src/app/shared/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit, AfterContentInit } from '@angular/core';
+import { ViewChild, ElementRef } from '@angular/core';
+import { fromEvent } from 'rxjs';
+
+@Component({
+  selector: 'app-carousel',
+  templateUrl: './carousel.component.html',
+  styleUrls: ['./carousel.component.scss']
+})
+
+export class CarouselComponent implements OnInit, AfterContentInit {
+  private currTranslate = 0;
+  private childrenLength: number;
+  public itemsPerPage;
+  public styleForContainer;
+  public itemWidth;
+
+  @ViewChild('slidesContainer') slidesContainer: ElementRef;
+
+
+  sizeHandler() {
+    if (innerWidth <= 424) {
+      this.itemWidth = '1000%';
+      this.itemsPerPage = 1;
+    } else if (innerWidth <= 768) {
+      this.itemWidth = '500%';
+      this.itemsPerPage = 2;
+    } else if (innerWidth <= 1024) {
+      this.itemWidth = '333.333333%';
+      this.itemsPerPage = 3;
+    } else {
+      this.itemWidth = '250%';
+      this.itemsPerPage = 4;
+    }
+  }
+
+  ngOnInit(): void {
+    this.sizeHandler();
+    const event = fromEvent(window, 'resize').subscribe(evt => {
```

Comment:
1 unsubscribe in ngOnDestory
2 add debounceTime ask @taras1313 for clarification